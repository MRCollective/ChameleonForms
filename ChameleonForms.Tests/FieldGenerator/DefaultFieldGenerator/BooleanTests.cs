using System.Web;
using ApprovalTests.Html;
using ChameleonForms.Component.Config;
using NUnit.Framework;

namespace ChameleonForms.Tests.FieldGenerator.DefaultFieldGenerator
{
    class BooleanTests : DefaultFieldGeneratorShould
    {
        [Test]
        public void Use_correct_html_for_single_checkbox_with_default_label()
        {
            var g = Arrange(m => m.RequiredBoolean);

            var result = g.GetFieldHtml(default(IFieldConfiguration));

            HtmlApprovals.VerifyHtml(result.ToHtmlString());
        }

        [Test]
        public void Use_correct_html_for_checked_single_checkbox_with_default_label()
        {
            var g = Arrange(m => m.RequiredBoolean, m => m.RequiredBoolean = true);

            var result = g.GetFieldHtml(default(IFieldConfiguration));

            HtmlApprovals.VerifyHtml(result.ToHtmlString());
        }

        [Test]
        public void Use_correct_html_for_single_checkbox_with_custom_label()
        {
            var g = Arrange(m => m.RequiredBoolean);

            var result = g.GetFieldHtml(new FieldConfiguration().InlineLabel("Some label"));

            HtmlApprovals.VerifyHtml(result.ToHtmlString());
        }

        [Test]
        public void Use_correct_html_for_boolean_list_and_false_value()
        {
            var g = Arrange(m => m.RequiredBoolean);

            var result = g.GetFieldHtml(new FieldConfiguration().AsRadioList());

            HtmlApprovals.VerifyHtml(result.ToHtmlString());
        }

        [Test]
        public void Use_correct_html_for_boolean_list_and_true_value()
        {
            var g = Arrange(m => m.RequiredBoolean, m => m.RequiredBoolean = true);

            var result = g.GetFieldHtml(new FieldConfiguration().AsRadioList());

            HtmlApprovals.VerifyHtml(result.ToHtmlString());
        }

        [Test]
        public void Use_correct_html_for_optional_boolean_list_and_no_value()
        {
            var g = Arrange(m => m.OptionalBooleanField);

            var result = g.GetFieldHtml(new FieldConfiguration().AsRadioList());

            HtmlApprovals.VerifyHtml(result.ToHtmlString());
        }

        [Test]
        public void Use_correct_html_for_optional_boolean_dropdown_and_no_value()
        {
            var g = Arrange(m => m.OptionalBooleanField);

            var result = g.GetFieldHtml(new FieldConfiguration().AsDropDown());

            HtmlApprovals.VerifyHtml(result.ToHtmlString());
        }

        [Test]
        public void Use_correct_html_for_optional_boolean_and_no_value()
        {
            var g = Arrange(m => m.OptionalBooleanField);

            var result = g.GetFieldHtml(new FieldConfiguration().AsDropDown());

            HtmlApprovals.VerifyHtml(result.ToHtmlString());
        }

        [Test]
        public void Use_correct_html_for_boolean_list_and_custom_labels()
        {
            var g = Arrange(m => m.RequiredBoolean);

            var result = g.GetFieldHtml(new FieldConfiguration().AsRadioList().WithTrueAs("True").WithFalseAs("False"));

            HtmlApprovals.VerifyHtml(result.ToHtmlString());
        }

        [Test]
        public void Use_correct_html_for_boolean_dropdown_list_and_false_value()
        {
            var g = Arrange(m => m.RequiredBoolean);

            var result = g.GetFieldHtml(new FieldConfiguration().AsDropDown());

            HtmlApprovals.VerifyHtml(result.ToHtmlString());
        }

        [Test]
        public void Use_correct_html_for_boolean_dropdown_list_and_true_value()
        {
            var g = Arrange(m => m.RequiredBoolean, m => m.RequiredBoolean = true);

            var result = g.GetFieldHtml(new FieldConfiguration().AsDropDown());

            HtmlApprovals.VerifyHtml(result.ToHtmlString());
        }

        [Test]
        public void Use_correct_html_for_nullable_required_boolean_checkbox_with_no_value()
        {
            var g = Arrange(m => m.RequiredNullableBoolean);

            var result = g.GetFieldHtml(default(IFieldConfiguration));

            HtmlApprovals.VerifyHtml(result.ToHtmlString());
        }

        [Test]
        public void Use_correct_html_for_nullable_required_boolean_list_with_no_value()
        {
            var g = Arrange(m => m.RequiredNullableBoolean);

            var result = g.GetFieldHtml(new FieldConfiguration().AsRadioList());

            HtmlApprovals.VerifyHtml(result.ToHtmlString());
        }

        [Test]
        public void Use_correct_html_for_nullable_required_boolean_dropdown_with_no_value()
        {
            var g = Arrange(m => m.RequiredNullableBoolean);

            var result = g.GetFieldHtml(new FieldConfiguration().AsDropDown());

            HtmlApprovals.VerifyHtml(result.ToHtmlString());
        }

        [Test]
        public void Use_correct_html_for_label_for_boolean_list()
        {
            var g = Arrange(m => m.RequiredNullableBoolean);

            var result = g.GetLabelHtml(new FieldConfiguration().AsRadioList());

            HtmlApprovals.VerifyHtml(result.ToHtmlString());
        }

        [Test]
        public void Use_correct_html_for_label_for_boolean_list_with_overridden_label()
        {
            var g = Arrange(m => m.RequiredNullableBoolean);

            var result = g.GetLabelHtml(new FieldConfiguration().AsRadioList().Label(new HtmlString("<strong>lol</strong>")));

            HtmlApprovals.VerifyHtml(result.ToHtmlString());
        }

        [Test]
        public void Use_correct_html_for_label_for_boolean_dropdown()
        {
            var g = Arrange(m => m.RequiredNullableBoolean);

            var result = g.GetLabelHtml(new FieldConfiguration().AsDropDown());

            HtmlApprovals.VerifyHtml(result.ToHtmlString());
        }

        [Test]
        public void Use_correct_html_for_label_for_boolean_checkbox()
        {
            var g = Arrange(m => m.RequiredNullableBoolean);

            var result = g.GetLabelHtml(new FieldConfiguration());

            HtmlApprovals.VerifyHtml(result.ToHtmlString());
        }
    }
}
