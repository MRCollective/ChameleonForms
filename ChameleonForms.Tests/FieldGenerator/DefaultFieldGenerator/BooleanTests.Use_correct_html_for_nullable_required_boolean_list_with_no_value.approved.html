<ul>
  <li>
    <label>
      <input class="input-validation-error" data-val="true" data-val-required="The RequiredNullableBoolean field is required." id="RequiredNullableBoolean_1" name="RequiredNullableBoolean" type="radio" value="true" /> Yes</label>
  </li>
  <li>
    <label>
      <input class="input-validation-error" id="RequiredNullableBoolean_2" name="RequiredNullableBoolean" type="radio" value="false" /> No</label>
  </li>
</ul>